certbot/certbot/crypto_util.py

```diff
@@ -174,18 +184,39 @@ def import_csr_file(csrfile, data):
     return PEM, util.CSR(file=csrfile, data=data_pem, form="pem"), domains
 
 
-def make_key(bits):
-    """Generate PEM encoded RSA key.
+def make_key(bits=1024, key_type="rsa", elliptic_curve=None):
+    """Generate PEM encoded RSA|EC key.
 
-    :param int bits: Number of bits, at least 1024.
+    :param int bits: Number of bits, at least 1024 for RSA.
 
-    :returns: new RSA key in PEM form with specified number of bits
-    :rtype: str
+    :param str ec_curve: The elliptic curve to use.
 
+    :returns: new RSA or ECDSA key in PEM form with specified number of bits
+              or of type ec_curve when key_type ecdsa is used.
+    :rtype: str
     """
-    assert bits >= 1024  # XXX
-    key = crypto.PKey()
-    key.generate_key(crypto.TYPE_RSA, bits)
+    if key_type.lower() == 'rsa':
```

Comment:
Any reason to lower the value? The message in the error raised in case of wrong value is quite explicit about `rsa` and `ecdsa` being the only acceptable values.

Reply:
Probably this was some of the remnants of the original PR. I'm not really familiar with zope, and how it works. I'll update it.